### ISSUE TITLE
Web shouldn't have a repeated DEX save every turn

### ISSUE THREAD

**Issue by Swofty** [SolastaMods/SolastaUnfinishedBusiness#3123]
The description of Web suggests that once a creature fails a DEX save, the only method of breaking free is through the Strength Check action.

> Each creature that enters them during its turn must make a Dexterity saving throw. On a failed save, the creature is restrained as long as it remains in the webs or until it breaks free. A creature restrained by the webs can use its actions to make a Strength check against your spell save DC. If it succeeds, it is no longer restrained.

The way it's currently implemented though, a restrained target will continue to make DEX saving throws on each of its turns (at disadvantage due to restrained). This creates a case where web can have zero effect on characters entering or starting their turn there for the first time.

1. Start turn
2. Enter web - Make DEX Saving throw - Fail
3. Another DEX saving throw is immediately triggered (not shown in logs?) - Success
4. Freely move on this same turn.

![image](https://user-images.githubusercontent.com/8569010/236159952-ada1b87e-b17b-43d8-a304-ef9f689116cb.png)

Additionally, I have no idea what's going on here, where enemies can make as many strength checks as they want against Web. (Should've been fixed according to #3037?)

![image](https://user-images.githubusercontent.com/8569010/236160254-3ead8228-5704-4094-9144-1e8591ce8218.png)

Attached repro has two files, one in combat and one outside of combat. With the in-combat one, you can step in and out of the web with the Fighter to check the behavior from the player side. After ending turn, you might be able to repro the enemy making multiple ability checks when it moves. You can repro the repeated saving throws in the out-of-combat save.

[web_repro.zip](https://github.com/SolastaMods/SolastaUnfinishedBusiness/files/11395756/web_repro.zip)

**Comment by ThyWoof** [comment]
Agree doing multiple strength checks is wrong. 

The dexterity saving is correct: Each creature that starts its turn in the webs or that enters them during its turn must make a Dexterity [saving throw](https://www.dndlounge.com/spell-save-dc-5e/). On a failed save, the creature is [restrained](https://www.dndlounge.com/restrained-5e/) as long as it remains in the webs or until it breaks free.

A creature restrained by the webs can use its action to make a Strength check against your spell save DC. If it succeeds, it is no longer restrained.

**Comment by ThyWoof** [comment]
“Until it breaks free”. Does it mean it should not do any more DEX savings after breaking free and still on web?

What if it leaves and enters again? Should reset and start all over again?

**Comment by ThyWoof** [comment]
![image](https://user-images.githubusercontent.com/13087646/236352730-cdf48b6a-7d51-4f8d-9e9b-13ea5df65f07.png)
was able to fix the repeating STR checks.

**Comment by Swofty** [comment]
> “Until it breaks free”. Does it mean it should not do any more DEX savings after breaking free and still on web?
> 
> What if it leaves and enters again? Should reset and start all over again?

If it's tabletop rules, then I believe yes, no DEX saving throws after breaking free but still on web. You can move on web without making a save. If you leave and then enter again on this turn, then you make a new save. If you break free but remain on the web at the end of your turn, then on the start of next turn you make a new DEX save.

I believe read as intended, once you fail a DEX save, you are restrained and make no more DEX saves until the restrained condition is ended. However, I think it is OK to make Solasta-unique rules to make a DEX save every turn. The issue though is that there seems to be two DEX saves when you start your turn in the web, when there should only be one.

Basically, there should not be this situation where the log says this back-to-back. 
1. [CREATURE] is now Restrained by Web
2. [CREATURE] is no longer Restrained by Web

![](https://user-images.githubusercontent.com/8569010/236159952-ada1b87e-b17b-43d8-a304-ef9f689116cb.png)

If the creature is restrained, it should be restrained until it uses an action to break it or at least be restrained until next turn where it makes a new DEX save.

**Comment by ThyWoof** [comment]
I’ll check if that sequence add / remove cond still happens after my changes

**Comment by ThyWoof** [comment]
![image](https://user-images.githubusercontent.com/13087646/236357088-cf283f96-f828-453b-ba4d-3ec47163c963.png)
I believe that sequence got fixed too. Can you try attached DLL on your end? replace the one inside the mod folder.
[SolastaUnfinishedBusiness.zip](https://github.com/SolastaMods/SolastaUnfinishedBusiness/files/11402529/SolastaUnfinishedBusiness.zip)

**Comment by Swofty** [comment]
It seems to be working now for targets starting inside the web. However if you start outside the web, and then walk onto it, it'll still trigger the multiple saves which results in the back-to-back restrained->unrestrained sequence.

**Comment by ThyWoof** [comment]
> It seems to be working now for targets starting inside the web. However if you start outside the web, and then walk onto it, it'll still trigger the multiple saves which results in the back-to-back restrained->unrestrained sequence.

very strange. this particular piece uses vanilla aura settings.

**Comment by ThyWoof** [comment]
found out why. I was copying from ConditionGrappledRestrainedRemorhaz which has special condition interruption on move.

fixed now. First Thug saved and got closer to my team. Second one was stopped at web edge on entering.
![image](https://user-images.githubusercontent.com/13087646/236364581-9fefc95d-7816-4ce5-9338-dcddf6d0b0f7.png)

please confirm on your end.
[SolastaUnfinishedBusiness.zip](https://github.com/SolastaMods/SolastaUnfinishedBusiness/files/11402849/SolastaUnfinishedBusiness.zip)
